stor/dx.py

```diff
@@ -171,22 +172,54 @@ def joinpath(self, *others):
         """
         return self.path_class(self.path_module.join(self.normpath(), *others))
 
+    def to_url(self):
+        """For compatibility with OBS - returns ``temp_url()``"""
+        return self.temp_url()
+
     def temp_url(self, lifetime=300, filename=None):
         """Obtains a temporary URL to a DNAnexus data-object.
 
+        If ``DX_FILE_PROXY_URL`` or ``[dx] file_proxy_url=`` is set, will use that to construct a
+        path instead, e.g.::
+
+            >>> stor.Path('dx://proj:/folder/mypath.csv').temp_url()
+            'https://dl.dnanex.us/F/D/awe1323/mypath.csv'
+            >>> with stor.settings.use({'dx': {'file_proxy_url':
+            ...     'https://myproxy.example.com/gateway'}):
+            ... stor.Path('dx://proj:/folder/mypath.csv').temp_url()
+            'https://my-dnax-proxy.example.com/gateway/proj/folder/mypath.csv'
+
+        The file proxy is assumed to be a service that, when given DX path and project, will proxy
+        through to DNAnexus to render content.
+
         Args:
             lifetime (int): The time (in seconds) the temporary
                 URL will be valid
             filename (str, optional): A urlencoded filename to use for
                 attachment, otherwise defaults to object name
+                (to bypass, set to empty string).
 
         Raises:
             ValueError: The path points to a project
-            NotFoundError: The path could not be resolved to a file
+            NotFoundError: The path could not be resolved to a file (only when ``file_proxy_url``
+                unset)
         """
         if not self.resource:
             raise ValueError('DX Projects cannot have a temporary download url')
+        file_proxy_url = stor.settings.get()['dx']['file_proxy_url']
```

Comment:
`stor.settings.get('dx', {}).get('file_proxy_url')`?

Reply:
this is always guaranteed to be a dictionary because of initialization. Also it'd be spelled `settings.get().get(dx', {}).get('file_proxy_url')`